flink-clients/src/main/java/org/apache/flink/client/deployment/ClusterDescriptor.java

```diff
@@ -44,5 +45,12 @@
 	 * @return Client for the cluster
 	 * @throws UnsupportedOperationException if this cluster descriptor doesn't support the operation
 	 */
-	ClientType deploy() throws UnsupportedOperationException;
+	ClientType deploySession() throws UnsupportedOperationException;
+
+	/**
+	 * Deploys a per-job cluster with the given job on the cluster.
+	 *
+	 * @return Cluster client to talk to the Flink cluster
+	 */
+	ClientType deployJob(final JobGraph jobGraph);
```

Comment:
I actually think that we should replace the unchecked exception by a checked exception. I will introduce one for this method. Changing the exception of `deploySession` is a follow up, then.